src/benchmark.cr

```diff
@@ -1,3 +1,47 @@
+# A Performance Benchmarking Library
+# Overview
+#
+# The Benchmark module provides methods for benchmarking Ruby code, giving
+# detailed reports on the time taken for each task.
+#
+
+# The Benchmark module provides methods to measure and report the time
+# used to execute Ruby code.
+#
+# * Measure the time to construct the string given by the expression
+#   <code>"a"*1_000_000_000</code>:
+#
+#       require 'benchmark'
+#
+#       puts Benchmark.measure { "a"*1_000_000_000 }
+#
+#   On my machine (OSX 10.10.3 on i7 2.2 Ghz) this generates:
+#
+#        0.190000   0.220000   0.410000 (  0.420185)
+#
+#   This report shows the user CPU time, system CPU time, the sum of
+#   the user and system CPU times, and the elapsed real time. The unit
+#   of time is seconds.
+
+# * Do some experiments sequentially using the #bm method:
+#
+#       require 'benchmark'
+#
+#       n = 5000000
+#       Benchmark.bm do |x|
+#         x.report { n.times do   ; a = "1"; end }
+#         x.report { 1.upto(n) do ; a = "1"; end }
+#       end
+#
+#   The result:
+#
+#              user     system      total        real
+#          1.000000   0.000000   1.000000 (  0.998261)
+#          0.980000   0.000000   0.980000 (  0.981335)
+#
+# * At present #bm(n) and #bmbm() are not provided
```

Comment:
I don't think there's always confusion. Compiling with `--release` takes a lot more time so you loose that "dynamic" feeling that you have when you just do `crystal foo.cr`. Previously it wasn't documented, but now there's a [chapter](http://crystal-lang.org/docs/using_the_compiler/index.html) in the docs. I want to wait a bit before changing this semantic.

There's also C. You know about `-O3`. Probably everyone knows that. But I can't remember anyone complaining about it. I think with time people learn it, and once you learn it you can't simply forget it.